include/suit_platform.h

```diff
@@ -27,109 +31,123 @@
 		struct zcbor_string *payload);
 
-/** Authenticate the given payload against the given signature.
+/** Authenticate the given manifest against the given signature.
  *
+ *  @param[in]  manifest_component_id  The manifest component ID, identifying
+ *                                     the type of manifest in the system.
  *  @param[in]  alg_id  The signature verification algorithm to use.
  *  @param[in]  key_id  The key to check the signature with.
  *  @param[in]  signature  The signature to check.
  *  @param[in]  data  The data that is signed by the @p signature.
  */
-int suit_plat_authenticate(enum suit_cose_alg alg_id, struct zcbor_string *key_id,
+int suit_plat_authenticate_manifest(struct zcbor_string *manifest_component_id,
+		enum suit_cose_alg alg_id, struct zcbor_string *key_id,
 		struct zcbor_string *signature, struct zcbor_string *data);
 
+/** Check that the provided manifest is allowed to be unsigned. */
+int suit_plat_authorize_unsigned_manifest(struct zcbor_string *manifest_component_id);
+
+#ifndef SUIT_PLATFORM_LEGACY_API_SUPPORT
 /** Check that the given component ID exists, is valid, and is authorized.
  *
  *  If so, create and return a component handle for it.
  *
  *  @param[in]  component_id  The CBOR-encoded component identifier.
- *  @param[in]  key_ids  The keys the current manifest was authenticated
- *                       against. This is to check that the manifest is
- *                       authorized to handle the given component ID.
- *  @param[in]  num_key_ids  The number of members in the @p key_ids list.
  *  @param[out] component_handle  A reference for use with other functions in
  *                                this API, instead of always passing the
  *                                @p parts.
  */
 int suit_plat_create_component_handle(struct zcbor_string *component_id,
-		struct zcbor_string *key_ids[SUIT_MAX_NUM_SIGNERS], size_t num_key_ids,
-		suit_component_t *component_handle);
+		suit_component_t *handle);
+#endif /* !SUIT_PLATFORM_LEGACY_API_SUPPORT */
 
-/** Check the provided payload against the provided digest */
-int suit_plat_check_image_match(enum suit_cose_alg alg_id,
-		struct zcbor_string *digest, size_t image_size,
-		suit_component_t image_handle);
+/** Release loaded component properties and handles assigned to them.
+ *
+ *  @param[in]  component_handle  The platform-specific component handle value.
+ */
+int suit_plat_release_component_handle(suit_component_t handle);
 
-/** Check whether the given slot is active for the given component. */
-int suit_plat_check_slot(suit_component_t component_handle, unsigned int slot);
+#ifndef SUIT_PLATFORM_LEGACY_API_SUPPORT
+/** Check the provided payload against the provided digest. */
+int suit_plat_check_image_match(suit_component_t handle,
+		enum suit_cose_alg alg_id, struct zcbor_string *digest,
+		size_t image_size);
+#endif /* !SUIT_PLATFORM_LEGACY_API_SUPPORT */
 
+/** Check the provided payload against the component value. */
```

Comment:
There's some inconsistency in the way functions are documented. Why not all functions have their arguments documented?

Reply:
Docs inside the `suit_platform.h` added.